data/mystery_event_msg.s

```diff
@@ -3,32 +3,92 @@
 
 	.section .rodata
 
-gUnknown_08674AE4:: @ 8674AE4
-	.incbin "baserom.gba", 0x674ae4, 0x32
+gText_MysteryGiftBerry:: @ 8674AE4
+	.string "Obtained a {STR_VAR_2} BERRY!\nDad has it at PETALBURG GYM.$"
 
-gUnknown_08674B16:: @ 8674B16
-	.incbin "baserom.gba", 0x674b16, 0x2c
+gText_MysteryGiftBerryTransform:: @ 8674B16
+	.string "The {STR_VAR_1} BERRY transformed into\none {STR_VAR_2} BERRY.$"
 
-gUnknown_08674B42:: @ 8674B42
-	.incbin "baserom.gba", 0x674b42, 0x28
+gText_MysteryGiftBerryObtained:: @ 8674B42
+	.string "The {STR_VAR_1} BERRY has already been\nobtained.$"
 
-gUnknown_08674B6A:: @ 8674B6A
-	.incbin "baserom.gba", 0x674b6a, 0x34
+gText_MysteryGiftSpecialRibbon:: @ 8674B6A
+	.string "A special RIBBON was awarded to\nyour party POKéMON.$"
 
-gUnknown_08674B9E:: @ 8674B9E
-	.incbin "baserom.gba", 0x674b9e, 0x36
+gText_MysteryGiftNationalDex:: @ 8674B9E
+	.string "The POKéDEX has been upgraded\nwith the NATIONAL MODE.$"
 
-gUnknown_08674BD4:: @ 8674BD4
-	.incbin "baserom.gba", 0x674bd4, 0x1c
+gText_MysteryGiftRareWord:: @ 8674BD4
+	.string "A rare word has been added.$"
 
-gUnknown_08674BF0:: @ 8674BF0
-	.incbin "baserom.gba", 0x674bf0, 0x12
+gText_MysteryGiftSentOver:: @ 8674BF0
+	.string "{STR_VAR_1} was sent over!$"
 
-gUnknown_08674C02:: @ 8674C02
-	.incbin "baserom.gba", 0x674c02, 0x2f
+gText_MysteryGiftFullParty:: @ 8674C02
+	.string "Your party is full.\n{STR_VAR_1} could not be sent over.$"
 
-gUnknown_08674C31:: @ 8674C31
-	.incbin "baserom.gba", 0x674c31, 0x55
+gText_MysteryGiftNewTrainer:: @ 8674C31
+	.string "A new TRAINER has arrived in\nHOENN.$"
+	.string "A new adversary has arrived in the\nBATTLE TOWER.$"
 
-gUnknown_08674C86:: @ 8674C86
-	.incbin "baserom.gba", 0x674c86, 0xd5a
+gText_MysteryGiftCantBeUsed:: @ 8674C86
+	.string "This data can’t be used in\nthis version.$"
+	.incbin "baserom.gba", 0x00674caf, 0x2e	//no pointer to this one
```

Comment:
Even if there's no pointer to it, if you know it's text you should still extract it.